mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityHelperSchemaValidationMethodGenerator.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.entity;
+
+import static com.datastax.oss.driver.api.mapper.annotations.SchemaHint.*;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import com.datastax.oss.driver.api.core.type.UserDefinedType;
+import com.datastax.oss.driver.api.mapper.annotations.SchemaHint;
+import com.datastax.oss.driver.internal.core.metadata.schema.DefaultTableMetadata;
+import com.datastax.oss.driver.internal.mapper.processor.MethodGenerator;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeName;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+
+public class EntityHelperSchemaValidationMethodGenerator implements MethodGenerator {
+
+  private final EntityDefinition entityDefinition;
+  private TypeElement entityTypeElement;
+
+  public EntityHelperSchemaValidationMethodGenerator(
+      EntityDefinition entityDefinition, TypeElement entityTypeElement) {
+    this.entityDefinition = entityDefinition;
+    this.entityTypeElement = entityTypeElement;
+  }
+
+  @Override
+  public Optional<MethodSpec> generate() {
+    MethodSpec.Builder methodBuilder =
+        MethodSpec.methodBuilder("validateEntityFields")
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(TypeName.VOID);
+
+    // get keyspaceId from context, and if not present fallback to keyspace set on session
+    methodBuilder.addStatement(
+        "$1T keyspaceId = context.getKeyspaceId() != null ? context.getKeyspaceId() : context.getSession().getKeyspace().orElse(null)",
+        CqlIdentifier.class);
+
+    // Handle case where keyspaceId = null. In such case we cannot infer and validate schema for
+    // table
+    // or udt
+    methodBuilder.beginControlFlow(
+        "if (!context.getSession().getMetadata().getKeyspace(keyspaceId).isPresent())");
+    methodBuilder.addStatement(
+        "return"); // todo maybe we should do log.warn("we cannot validate schema because keyspace
```

Comment:
should we add `warn` for that case or fail-silently?

Reply:
I think we should warn: if `schemaValidationEnabled == true` and the validation can't run, we should let the user know.